# MAG


# Figure 1 Quality control of the raw reads: 

BGR130305 R1 (left) and R2 (right)

<img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130305_mapped_R1.png" width="500"/> <img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130305_mapped_R2.png" width="500"/>


BGR 130527 R1 (left) R2 (right)

<img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130527_mapped_R1.png" width="500"/> <img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130527_mapped_R2.png" width="500"/>

BGR 130708 R1 (left) R2 (right)

<img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130708_mapped_R1.png" width="500"/> <img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/BGR_130708_mapped_R2.png" width="500"/>

## Fastp report 

<img src="https://github.com/MarieRaasch/biol217/blob/main/Ressources/Fastp report.png" width="500"/>

## stats of contigs database after contigs data preparation 

(https://github.com/MarieRaasch/biol217/blob/main/Ressources/Contigs%20DB%20Stats.pdf)
(https://github.com/MarieRaasch/biol217/blob/main/Ressources/Contigs%20DB%20Stats%20Archaea.pdf)


# Pangenomics

## Table 1 Bacterial genomes used for the pangenome analysis with anvi'o

| Classification         | NCBI ID        |
|------------------------|----------------|
| Clostridium botulinum  | NC_009495.1    |
| Clostridium perfringens| NZ_CP075979.1  |
| Chlostridium tetani    | NC_004557.1    |
| Pseudomonas putida     | NC_021505.1    |
| Vibrio cholerae        | NZ_CP043554.1  |
| Vibrio harveyi         | NZ_CP125875.1  |
| Aliivibrio fischeri    | NC_011184.1    |
| Vibrio anguillarum     | NZ_CP031479.1  |
| Vibrio parahaemolyticus| NC_004603.1    |
| Vibrio coralliilyticus | NZ_CP063051.1  |
